app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java

```diff
@@ -1409,7 +1413,10 @@ public void run() {
 
     @Override
     public void onGlobalFocusChanged(View oldFocus, View newFocus) {
-        updateFocusedView(newFocus);
+        if (mExternalFocusedView == newFocus)
+            updateFocusedView(mAttachedWindow);
+        else
+            updateFocusedView(newFocus);
```

Comment:
We can simplify this to `updateFocusedView(mExternalFocusedView == newFocus ? mAttachedWindow : newFocus)`

Reply:
Done